packages/auth-providers/dbAuth/api/src/__tests__/DbAuthHandler.test.js

```diff
@@ -2340,6 +2340,41 @@ describe('dbAuth', () => {
       expect.assertions(2)
     })
 
+    it('createUser db check is called with insensitive string when user has provided one in SignupFlowOptions', async () => {
+      const spy = jest.spyOn(db.user, 'findUnique');
+      options.signup.usernameMatch = "insensitive"
+
+      const dbUser = await createDbUser()
+      event.body = JSON.stringify({
+        username: dbUser.email,
+        password: 'password',
+      })
+      const dbAuth = new DbAuthHandler(event, context, options)
+
+      dbAuth._createUser();
+      expect(spy).toHaveBeenCalled()
+      expect(spy).toHaveBeenCalledWith({ 'where' : {
+        'email' : expect.objectContaining({ mode: 'insensitive'})
+      } })
+    })
+
```

Comment:
@cannikin Hey! Thanks for stopping by the stream earlier. 
I have got this implemented and tests are written. However. I am getting a concurrency issue from the spy I believe. The second test, even though I am deleting the options.signup.usernameMatch the test is failing and showing the insensitive option being defined. 

Thoughts?
![image](https://user-images.githubusercontent.com/1301564/230488397-bec0654e-ce22-46b5-a7c4-e27b7df30124.png)

Reply:
Hmm, is there a `jest.clearAllMocks()` in a `beforeEach()` or `afterEach()`? Not sure if that matters with spies, though.

Reply:
Hey @cannikin that worked, I have now run all tests, linter etc as well